module_analysis/1_calc_module_composition.py

```diff
@@ -0,0 +1,283 @@
+)
+
+pao1_gene_group_composition.head()
+
+
+def label_modules(
+    module_ids_list, membership_df, core_genes_list, acc_genes_list, out_df
+):
+    all_genes = list(membership_df.index)
+
+    for module_id in module_ids_list:
+        # Find genes in module and outside module
+        genes_in_module = list(
+            membership_df[membership_df["module id"] == module_id].index
+        )
+        genes_outside_module = list(set(all_genes).difference(genes_in_module))
+
+        # Get core and accessory in module
+        core_genes_in_module = [
+            gene for gene in genes_in_module if gene in core_genes_list
+        ]
+        acc_genes_in_module = [
+            gene for gene in genes_in_module if gene in acc_genes_list
+        ]
+
+        # Get core and accessory genes outside of module
+        core_genes_outside_module = [
+            gene for gene in genes_outside_module if gene in core_genes_list
+        ]
+        acc_genes_outside_module = [
+            gene for gene in genes_outside_module if gene in acc_genes_list
+        ]
+
+        # Check
+        assert len(all_genes) == len(core_genes_in_module) + len(
+            acc_genes_in_module
+        ) + len(core_genes_outside_module) + len(acc_genes_outside_module)
+        assert len(genes_in_module) == len(core_genes_in_module) + len(
+            acc_genes_in_module
+        )
+
+        # Make contingency table
+        # -----|inside module|outside module
+        # core | # genes     | # genes
+        # acc  | # genes     | # genes
+        observed_contingency_table = np.array(
+            [
+                [len(core_genes_in_module), len(core_genes_outside_module)],
+                [len(acc_genes_in_module), len(acc_genes_outside_module)],
+            ]
+        )
+
+        # H0: The probability that the gene is core is the same
+        # whether or not you're in the module or outside
+        # H1: The probability that a gene is core is higher or lower inside the module
+        # than outside the module
+        # Should I do 2 one-sided tests, one for testing if
+        odds_ratio, pval = scipy.stats.fisher_exact(observed_contingency_table)
+
+        # odds_ratio, pval_greater = scipy.stats.fisher_exact(
+        #    observed_contingency_table, alternative="greater"
+        # )
+        # odds_ratio, pval_less = scipy.stats.fisher_exact(
+        #    observed_contingency_table, alternative="less"
+        # )
+
+        # Fill in df
+        out_df.loc[module_id, "num core"] = len(core_genes_in_module)
+        out_df.loc[module_id, "num acc"] = len(acc_genes_in_module)
+        out_df.loc[module_id, "odds ratio"] = odds_ratio
+        out_df.loc[module_id, "p-value"] = pval
+
+        # Do we want to include p-value criteria?
+        # Given the small sizes of the modules I think this is why
+        # most of the findings are not significant
+        # Use bonferrroni corrected p-value
```

Comment:
If you're finding that this correction is too stringent, it would probably be fine to do a FDR correction here, since you're probably okay with having a few false positives. You can do this using [statsmodels](https://www.statsmodels.org/dev/generated/statsmodels.stats.multitest.multipletests.html) - I use the `fdr_bh` correction there in my work.

Reply:
This is a good point. I'll add a params for this